py/server/deephaven/experimental/s3.py

```diff
@@ -84,6 +91,36 @@ def __init__(self,
             num_concurrent_write_parts (int): the maximum number of parts that can be uploaded concurrently when writing
                 to S3 without blocking, defaults to 64. Setting a higher value may increase throughput, but may also
                 increase memory usage.
+            profile_name (str): the profile name used for configuring the default region, credentials, etc., when
```

Comment:
This documentation is significantly improved over prior versions, but I have read it multiple times and find it very hard to fully grasp.  I feel like there are a lot of details here that will make complete sense if you already understand the answer, but there is big picture stuff missing so I never feel like I have a high-level framework to interpret the docs.

I would like @jjbrosnan to comment, since he has to document and support such things.

Reply:
While I don't see any major issues, it does feel weirdly worded to me. Here's my attempt at a rework

```
profile_name (str): the profile name used to configure the default region and credentials for reading and writing to S3. Credentials are assumed to be provided as part of the profile when this argument is used. If credentials are not part of the profile, set them with the access_key_id and secret_access_key input arguments. If no profile is given, the AWS SDK picks the profile from the "aws.profile" system property, the "AWS_PROFILE" environment variable, or defaults to "default".
```

Reply:
Hi @jjbrosnan, thanks for the comment. Can you pleas have a look at the pydocs for all the fields in this file?

Reply:
So, I looked over each one of the args. For the most part, they're good. Here are my key thoughts.

- Formatting is inconsistent. Some do:

```
arg (type): blah blah
```

Others are like this:

```
arg (type):
    blah blah
```

- Is the int fragment size given in KiB as well? Only the default is said to be in KiB, but this isn't specified.
- I think I know what a DurationLike is, but will a reader?
- `connection_timeout` and `read_timeout` should split the first sentence at the first comma.
- The docstrings for `access_key_id` and `secret_access_key` don't mention any of the other args that like `profile_name`. I don't know if they actually need to, but there's just a lot of overlap.
- `write_part_size` and `num_concurrent_write_parts` use `parts` instead of `chunks`. Is there a reason for this?
- `write_part_size` uses an exact number of bytes, then a MiB equivalence in parentheses. This one is a bit too verbose.

```
write_part_size (int): The chunk size when writing to S3. The default is 10 MiB. The minimum allowed chunk size is 5 MiB. Higher chunk size may increase throughput but also increase memory usage. Writing to S3 can be done in a maximum of 10,000 chunks, so the maximum size of a single file that can be written is about 98 GiB for the default chunk size.
```

- Some of these args are (I think) specific to AWS, but we've specifically called out S3-compatible APIs in the past. It's a bit hard to distinguish which args are specific to AWS S3 and which are not.
- Why would you want to use either of the last two args if you just specify a `profile_name`, `config_file_path`, or `credentials_file_path`? Do you have to set one of these when you use one or more of those?

This is a lot of stuff. If you want me to elaborate on a specific point, I'd be happy to.

Reply:
Hi JJ, thanks for the detailed review, I can respond to each of your points.

- Fixed the inconsistent formatting
- Clarified that the fragment size is in bytes
- I have specified examples in the pydoc. I think that's the best we cna do here.
- Updated
-  access_key_id and secret_access_key are not related to profile_name, so different pydocs.
- This is S3 terminology, they call it parts on write side instead of chunks. So we have used parts in the API and docs.
- Updated the write_part_size doc
- All of these args are S3 specific. Mentioned in the Description for the class.
- So if a user specifies profile_name, config_file_path, or credentials_file_path, then specifying use_profile_credentials would provide no additional advantage. But if a user has not specified  these three parameters, by default, we have a whole list of places which are checked for discovering credentials. Now if a user only wants to check the profile files, then they can set use_profile_credentials. 
Note that these are more technical and debugging related parameters where we want to override the more accepting default behavior of searching for credentials everywhere and enforcing a particular plocation.